azure-servicebus/src/test/java/com/microsoft/azure/servicebus/TestUtils.java

```diff
@@ -27,20 +35,30 @@
 			System.err.println(NAMESPACE_CONNECTION_STRING_ENVIRONMENT_VARIABLE_NAME + " environment variable not set. Tests will not be able to connect to to any service bus entity.");
 		}
 		namespaceConnectionStringBuilder = new ConnectionStringBuilder(namespaceConnectionString);
+
+		// Read proxy settings - proxy settings can still be set even if tests are not set to run with a proxy
+        runWithProxy = Boolean.valueOf(System.getenv(RUN_WITH_PROXY_ENV_VAR));
+        proxyHostName = System.getenv(PROXY_HOSTNAME_ENV_VAR);
+        proxyPort = System.getenv(PROXY_PORT_ENV_VAR) == null ?
+                        0 : Integer.valueOf(System.getenv(PROXY_PORT_ENV_VAR));
 	}
 	
 	public static URI getNamespaceEndpointURI()
     {
         return namespaceConnectionStringBuilder.getEndpoint();
     }
-    
-    public static ClientSettings getClientSettings()
-    {
-        if (Boolean.valueOf(System.getenv(RUN_WITH_PROXY_ENV_VAR))) {
-            return TestUtils.getProxyClientSettings();
-        } else {
-            return Util.getClientSettingsFromConnectionStringBuilder(namespaceConnectionStringBuilder);
+
+    public static String getNamespaceConnectionString() { return namespaceConnectionString; }
```

Comment:
>getNamespaceConnectionString [](start = 25, length = 28)

Probably not required,as tests using this method are also not required.

Reply:
@yvgopal Its used in ProxySelectorTests